src/validation.cpp

```diff
@@ -5183,6 +5185,296 @@ CChainState& ChainstateManager::InitializeChainstate(CTxMemPool& mempool, const
     return *to_modify;
 }
 
+Optional<AssumeutxoData> ExpectedAssumeutxo(const int height, const CChainParams& chainparams)
```

Comment:
This is not performance critical I think. `const` has the additional benefit over being easier to reason about.

Reply:
Yes, there have been a few conversations about this recently, like https://github.com/bitcoin/bitcoin/pull/19845#discussion_r489776345 and others. The developer notes mention referring to the [C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/) but maybe a line about this in the notes would save everyone time.

Reply:
@sipa these C++ Core Guidelines appear to recommend not passing cheaply copied values as const:
- https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#con-constants-and-immutability
- https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Rf-in
- https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#f15-prefer-simple-and-conventional-ways-of-passing-information

Reply:
The only relevant discussion about by-value function arguments declared const I can see is:

> Exception
>
> Function arguments are rarely mutated, but also rarely declared const. To avoid confusion and lots of false positives, don’t enforce this rule for function arguments.

So I read this as: the general advice (make immutable things const) applies, but you're not supposed to warn about it for function arguments because that would confuse people.

I consider this personal style, and up to the author of the code.

Reply:
@sipa per the replies by you and @vasild at https://github.com/bitcoin/bitcoin/pull/19845#discussion_r490289235, *My take is the same as the isocpp guidelines - no const for function arguments that are passed by value*, I had the impression that this is considered a guideline, but I won't comment on it further in reviews.